test/resauce/core_test.clj

```diff
@@ -1,14 +1,216 @@
 (ns resauce.core-test
-  (:require [clojure.test :refer :all]
+  (:require [babashka.fs :as fs]
+            [clojure.test :refer :all]
             [clojure.java.io :as io]
-            [resauce.core :refer :all]))
+            [clojure.string :as str]
+            [resauce.core :refer :all])
+  (:import [java.io File]
+           [java.net URI URL]
+           [java.nio.file Path]))
+
+(defn- str->uri [^String s]
+  (when s
+    (try
+      (.toURI (io/as-url s))
+      (catch Exception _
+        (.toUri (fs/path s))))))
+
+;;; FS Files/Dirs
+
+(defn- prefixed-w-file? [f]
+  (str/starts-with? (str f) "file:"))
+
+(defn- prefix-with-file [file-path]
+  (str "file:" file-path))
+
+(defn- strip-file-prefix [f]
+  (subs (str f) (count "file:")))
+
+(defn- ->file-reprs
+  "File | Path | URL | URI | \"...\" -> [File
+                                         Path
+                                         URL
+                                         URI
+                                         \"...\"
+                                         \"file:...\"]"
+  [f]
+  ((cond
+     (instance? File f)
+     (juxt identity
+           fs/path
+           io/as-url
+           (memfn ^File toURI)
+           str
+           prefix-with-file)
+
+     (instance? Path f)
+     (juxt fs/file
+           identity
+           (comp io/as-url fs/file)
+           (memfn ^Path toUri)
+           str
+           prefix-with-file)
+
+     (instance? URL f)
+     (juxt fs/file
+           fs/path
+           identity
+           (memfn ^URL toURI)
+           strip-file-prefix
+           str)
+
+     (instance? URI f)
+     (juxt fs/file
+           fs/path
+           (memfn ^URI toURL)
+           identity
+           strip-file-prefix
+           str)
+
+     (string? f)
+     (juxt fs/file
+           fs/path
+           (comp (memfn ^URI toURL) str->uri)
+           str->uri
+           (if (prefixed-w-file? f) strip-file-prefix identity)
+           (if (prefixed-w-file? f) identity prefix-with-file)))
+   f))
+
+(defn some-file-path ^Path []
+  (->> (fs/list-dir (fs/home))
+       (filter fs/regular-file?)
+       (first)))
+
+(defn some-file-url ^URL []
+  ;; NB: Use some concrete resource that we know lives on the classpath as file.
+  (io/resource "resauce/core.clj"))
```

Comment:
An already used `"resauce/core.clj"` file is complemented by any file from the home directory.